### ISSUE TITLE
ContentType needs to be set on the HttpEntity, not on the Response header.

### ISSUE THREAD

**Issue by nathan-gs** [NET-A-PORTER/pre-canned#8]
Spray changed the way it handles ContentType, it is required for the ContentType to be set on the HttpEntity.

Calling:
```
httpMock expect get and path("/test") and respond using resource('/test.json') and contentType(ContentTypes.`application/json`) end()
```

Will result in a warning and the incorrect or missing content type being set:
```
[WARN] [03/20/2015 09:44:50.452] [TenantApiClientSuite-akka.actor.default-dispatcher-10] [akka://TenantApiClientSuite/user/IO-HTTP/listener-0/0] Explicitly set response header 'Content-Type: application/json; charset=UTF-8' is ignored, the response Content-Type is set via the response's HttpEntity!
```

**Comment by theon** [comment]
Ah, thanks for the heads up!

**Comment by theon** [comment]
Published under version `0.0.6`
